File: my-folder/1915-check-if-one-string-swap-can-make-strings-equal/solution.py
class Solution:
    def areAlmostEqual(self, s1: str, s2: str) -> bool:
        counter1 = Counter(s1)
        counter2 = Counter(s2)
        if counter1!=counter2: return False
        i = 0
        count = 0
        while i<len(s1):
            if s1[i]!=s2[i]:
                count+=1
            if count>2: return False
            i+=1
        return True

File: my-folder/1915-check-if-one-string-swap-can-make-strings-equal/solution.cpp
class Solution {
public:
    bool areAlmostEqual(string s1, string s2) {
        unordered_map<char,int> c1;
        unordered_map<char,int> c2;


        for (char ch:s1)
        {
            c1[ch]+=1;
        }
        for (char ch:s2)
        {
            c2[ch]+=1;
        }
        if (c1!=c2) return false;
        int count = 0;
        for (int i=0; i<min(s1.size(),s2.size()); i++)
        {
            if (s1[i]!=s2[i]) count+=1;
            if (count>2) return false;
        }
        // for (const auto& pair:c1){
        //     cout<<"( "<<pair.first<<", "<<pair.second<<" )"<<endl;
        // }
        return true;


    }
};
